geniepy/src/geniepy/datamgmt/parsers.py

```diff
@@ -215,6 +215,30 @@ def parse(data, dtype=DataType.DF) -> DataFrame:
         except:
             return None
 
+            
+class CitationParser(BaseParser):
+    """Implementation of Scientific Journal Ratings Parser."""
+
+    default_type: DataType = DataType.DF
+    scraper: gs.CitationScraper = gs.CitationScraper()
+    schema: Schema = Schema(
+        [
+            Column("pmid", [IsDtypeValidation(np.int64)]),
+            Column("citation_count", [IsDtypeValidation(np.int64)]),
+            Column("citation_pmid"),
+        ]
+    )    
+
+    @staticmethod
+    def parse(data, dtype=DataType.DF) -> DataFrame:
+        """Parse data and convert according to parser schema."""
+        try:
+            parsed_df = DataFrame(data)
+            parsed_df.columns = ["pmid", "citation_count", "citation_pmid"]
```

Comment:
for future reference, this generates a view and causes a warning from pandas. You should use .copy() to create a dataframe instead of a view.